lib/SIL/SILInstructions.cpp

```diff
@@ -2484,6 +2486,31 @@ DestructureTupleInst *DestructureTupleInst::create(SILModule &M,
 }
 
 // SWIFT_ENABLE_TENSORFLOW
+GraphOperationInst::GraphOperationInst(
+    SILModule &M, SILDebugLocation loc, Identifier name,
+    ArrayRef<SILValue> arguments, ArrayRef<GraphOperationAttribute> attrs,
+    ArrayRef<SILType> resultTypes,
+    ArrayRef<ValueOwnershipKind> resultOwnerships) :
+  InstructionBase(loc),
+  MultipleValueInstructionTrailingObjects(this, resultTypes,
+                                          resultOwnerships),
+  Name(name), NumOperands(arguments.size())
+{
+  auto allOperands = getAllOperands();
+  for (unsigned i : indices(arguments))
+    new (&allOperands[i]) Operand(this, arguments[i]);
+  auto attrBuf = AlignedAlloc(sizeof(GraphOperationAttribute) * attrs.size(),
```

Comment:
@lattner 
Could you please verify that I allocated `Attributes` optimally here?
As we discussed, I did this approach rather than `Attributes(M.allocateCopy(attrs))`.

Reply:
I'd suggest just doing attrBuf = new GraphOperationAttribute[attrs.size()];

then "delete[] ptr" in the dtor.